Makefile

```diff
@@ -1,156 +1,89 @@
-ifdef B_BASE
-include $(B_BASE)/common.mk
-include $(B_BASE)/rpmbuild.mk
-REPO=$(call git_loc,xen-api)
-else
-MY_OUTPUT_DIR ?= $(CURDIR)/output
-MY_OBJ_DIR ?= $(CURDIR)/obj
-REPO ?= $(CURDIR)
-
-RPM_SPECSDIR?=$(shell rpm --eval='%_specdir')
-RPM_SRPMSDIR?=$(shell rpm --eval='%_srcrpmdir')
-RPM_SOURCESDIR?=$(shell rpm --eval='%_sourcedir')
-RPMBUILD?=rpmbuild
-XEN_RELEASE?=unknown
-endif
-
-BASE_PATH=$(shell scripts/base-path scripts/xapi.conf)
-
-JQUERY=$(CARBON_DISTFILES)/javascript/jquery/jquery-1.1.3.1.pack.js
-JQUERY_TREEVIEW=$(CARBON_DISTFILES)/javascript/jquery/treeview/jquery.treeview.zip
-
-COMPILE_NATIVE ?= yes
-COMPILE_BYTE ?= no
-export COMPILE_NATIVE COMPILE_BYTE
-
 include config.mk
+INSTALL_PATH = $(DESTDIR)/$(shell ocamlfind printconf destdir)
+OCAMLFIND_DESTDIR = $(INSTALL_PATH)
+CONFIGUREFLAGS=--enable-tests
 
-OCAMLPATH=
-EXTRA_INSTALL_PATH=
-
-export ETCDIR OPTDIR PLUGINDIR HOOKSDIR INVENTORY VARPATCHDIR LIBEXECDIR XAPICONF SCRIPTSDIR SHAREDIR WEBDIR CLUSTER_STACK_ROOT BINDIR SBINDIR UDEVDIR OCAMLPATH EXTRA_INSTALL_PATH
-export DISABLE_WARN_ERROR
-
-.PHONY: all
-all: version
-	omake -j 8 phase1
-	omake -j 8 phase2
-	omake -j 8 phase3
-
-.PHONY: phase1 phase2 phase3
-phase1: version
-	omake phase1
-phase2:
-	omake phase2
-phase3:
-	omake phase3
-
-.PHONY: test
-test:
-	omake test
-	@echo @
-	@echo @ Running unit tests
-	@echo @
-#	Pipe ugly bash output to /dev/null
-	@echo @ xapi unit test suite
-	@./ocaml/xapi/suite -verbose true -shards 1
-	@echo
-	@echo @ HA binpack test
-	@./ocaml/xapi/binpack
-	./ocaml/database/database_test
-	./ocaml/license/test_daily_license_check -verbose true
-#	The following test no longer compiles:
-#	./ocaml/xenops/device_number_test
-#	The following test must be run in dom0:
-#	./ocaml/xenops/cancel_utils_test
-
-.PHONY: install
-install:
-	omake install
-	omake lib-uninstall
-	omake lib-install
-
-.PHONY: lib-install
-lib-install:
-	omake DESTDIR=$(DESTDIR) lib-install
-
-.PHONY: lib-uninstall
-lib-uninstall:
-	omake DESTDIR=$(DESTDIR) lib-uninstall
-
-.PHONY: noarch-install
-noarch-install: doc
-	omake noarch-install
-
-.PHONY: clean
-clean:
-	omake clean
-	omake lib-uninstall
-	rm -rf dist/staging
-	rm -f .omakedb .omakedb.lock xapi.spec xapi_version.ml
-	find -name '*.omc' -delete
-
-.PHONY: otags
-otags:
-	otags -vi -r . -o tags
-
-.PHONY: doc-html
-doc-html: phase1
-	omake doc-html
-
-.PHONY: doc-pdf
-doc-pdf: phase1
-	omake doc-pdf
-
-.PHONY: doc-json
-doc-json: phase1
-	omake doc-json
-
-.PHONY: doc
-doc: doc-html doc-pdf doc-json
```

Comment:
All gone - it'll have to come back for the API docs on github, but we'll fix that next